builder/vsphere/driver/vm_cdrom.go

```diff
@@ -55,19 +55,24 @@ func (vm *VirtualMachineDriver) CreateCdrom(c *types.VirtualController) (*types.
 	return device, nil
 }
 
-func (vm *VirtualMachineDriver) RemoveCdroms() error {
+func (vm *VirtualMachineDriver) RemoveCdroms(nCdroms int) error {
```

Comment:
Working through this I see that if nCdroms is greater than 0, then nCdroms will be removed. You do this by subslicing the cdroms slice which will remove the devices from 0 up until nCdroms-1. But if you nCdroms is 0 all devices get removed. That is not the behavior I would expect. 

Instead of changing the function signature for RemoveCdroms I suggest creating a new method called RemoveVirtualCdromsN(int)

```
//RemoveVirtualCdromsN removes up to n CDRoms from the associated vm.
//An error will occur If n is larger then the attached CDRoms. 
//n == 0 results in no CDRoms being removed
func (vm *VirtualMachineDriver) RemoveVirtualCdromsN(n int) error {
        if n == 0 {
		return nil
	}	
        devices, err := vm.Devices()
	if err != nil {
		return err
	}
	cdroms := devices.SelectByType((*types.VirtualCdrom)(nil))
        if (n < 0) || (n > len(cdroms)){
         return errors.New("invalid number: n must be between 0 and %d, len(cdroms))
        }
       
        cdroms = cdroms[:n]
	if err = vm.RemoveDevice(true, cdroms...); err != nil {
		return err
	}

//// I don know what needs to happen with the 
// sata := devices.SelectByType((*types.VirtualAHCIController)(nil)) ....
	return nil
}

Reply:
Sure, will do

Reply:
Done. I noticed though that we already have `RemoveCdroms` without infix `Virtual`, which makes sense because typically the plugin is dealing with "virtual hardware", not a passthrough. So I took the liberty of naming it a `RemoveNCdroms`, i.e. without the `Virtual` infix.

Tell if you want that to be changed.